### ISSUE TITLE
4D-API_cs2cs-style.gie fails on Debian unstable (with GCC 8)

### ISSUE THREAD

**Issue by sebastic** [OSGeo/PROJ#1084]
`4D-API_cs2cs-style.gie` started to fail on Debian unstable recently. Quite possibly related to GCC 8 becoming the default compiler.

```
-------------------------------------------------------------------------------
Reading file 'GDA.gie'
-------------------------------------------------------------------------------
total:  3 tests succeeded,  0 tests skipped,  0 tests failed.
-------------------------------------------------------------------------------
PROJ_LIB=../../nad ../../src/gie builtins.gie
-------------------------------------------------------------------------------
Reading file 'axisswap.gie'
-------------------------------------------------------------------------------
total: 27 tests succeeded,  0 tests skipped,  0 tests failed.
-------------------------------------------------------------------------------
PROJ_LIB=../../nad ../../src/gie deformation.gie
-------------------------------------------------------------------------------
Reading file '4D-API_cs2cs-style.gie'
     -----
     FAILURE in 4D-API_cs2cs-style.gie(114):
     expected: 12.5 55.5 0
     got:      inf   inf   inf
     deviation:  999999999.999000 mm,  expected:  150.000000 mm
-------------------------------------------------------------------------------
total: 28 tests succeeded,  0 tests skipped,  1 tests FAILED!
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
Reading file 'deformation.gie'
-------------------------------------------------------------------------------
total: 37 tests succeeded,  0 tests skipped,  0 tests failed.
-------------------------------------------------------------------------------
make[5]: *** [Makefile:455: 4D-API-cs2cs-style] Error 1
make[5]: *** Waiting for unfinished jobs....
-------------------------------------------------------------------------------
Reading file 'builtins.gie'
-------------------------------------------------------------------------------
total: 1568 tests succeeded,  0 tests skipped,  0 tests failed.
-------------------------------------------------------------------------------
```

**Comment by rouault** [comment]
@sebastic. Fixed. I suspect a gcc optimization bug (reported as https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86914 )

**Comment by sebastic** [comment]
Thanks for the patch and in-depth analysis. I've included the changes from 143b4d3 as a patch in the 5.1.0 Debian package, but the test still fails in the same way:
```
make[5]: Entering directory '/build/proj-5.1.0/test/gie'
PROJ_LIB=../../nad ../../src/gie 4D-API_cs2cs-style.gie
PROJ_LIB=../../nad ../../src/gie GDA.gie
PROJ_LIB=../../nad ../../src/gie axisswap.gie
-------------------------------------------------------------------------------
Reading file 'GDA.gie'
-------------------------------------------------------------------------------
total:  3 tests succeeded,  0 tests skipped,  0 tests failed.
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
Reading file 'axisswap.gie'
-------------------------------------------------------------------------------
total: 27 tests succeeded,  0 tests skipped,  0 tests failed.
-------------------------------------------------------------------------------
PROJ_LIB=../../nad ../../src/gie builtins.gie
PROJ_LIB=../../nad ../../src/gie deformation.gie
-------------------------------------------------------------------------------
Reading file 'deformation.gie'
-------------------------------------------------------------------------------
total: 37 tests succeeded,  0 tests skipped,  0 tests failed.
-------------------------------------------------------------------------------
PROJ_LIB=../../nad ../../src/gie ellipsoid.gie
-------------------------------------------------------------------------------
Reading file '4D-API_cs2cs-style.gie'
     -----
     FAILURE in 4D-API_cs2cs-style.gie(114):
     expected: 12.5 55.5 0
     got:      inf   inf   inf
     deviation:  999999999.999000 mm,  expected:  150.000000 mm
-------------------------------------------------------------------------------
total: 28 tests succeeded,  0 tests skipped,  1 tests FAILED!
-------------------------------------------------------------------------------
make[5]: *** [Makefile:458: 4D-API-cs2cs-style] Error 1
make[5]: *** Waiting for unfinished jobs....
-------------------------------------------------------------------------------
Reading file 'ellipsoid.gie'
-------------------------------------------------------------------------------
total: 30 tests succeeded,  0 tests skipped,  0 tests failed.
-------------------------------------------------------------------------------
-------------------------------------------------------------------------------
Reading file 'builtins.gie'
-------------------------------------------------------------------------------
total: 1568 tests succeeded,  0 tests skipped,  0 tests failed.
-------------------------------------------------------------------------------
make[5]: Leaving directory '/build/proj-5.1.0/test/gie'
```

Full buildlog: [proj_5.1.0-2_amd64.txt](https://github.com/OSGeo/proj.4/files/2280104/proj_5.1.0-2_amd64.txt)

**Comment by rouault** [comment]
@sebastic  Weird. Can you just try to build master and make check (after installing egm96_15.gtx in nad/) with the same compiler ?

**Comment by sebastic** [comment]
`make check` on `master` has two failures:
```
make[3]: Entering directory '/home/bas/git/OSGeo/proj/test/gie'
PROJ_LIB=../../nad ../../src/gie 4D-API_cs2cs-style.gie
-------------------------------------------------------------------------------
Reading file '4D-API_cs2cs-style.gie'
     -----
     FAILURE in 4D-API_cs2cs-style.gie(114):
     expected: 12.5 55.5 0
     got:      inf   inf   inf
     deviation:  999999999.999000 mm,  expected:  150.000000 mm
     -----
     FAILURE in 4D-API_cs2cs-style.gie(256):
     expected: -100.0004058367 40.0000058947 0.0000
     got:      -100.000000000000   40.000000000000
     deviation:  34662.139691 mm,  expected:  200.000000 mm
-------------------------------------------------------------------------------
total: 35 tests succeeded,  2 tests skipped,  2 tests FAILED!
-------------------------------------------------------------------------------
make[3]: *** [Makefile:466: 4D-API-cs2cs-style] Error 2
```

**Comment by rouault** [comment]
OK, I can reproduce within a Debian unstable VM with gcc 8.2. Regarding the first failure initially reported (4D-API_cs2cs-style.gie(114)), this is a difference between gcc 8.1 and 8.2, and the later only generates the correct code if using the [] syntax (I've updated https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86914 with it). Investigating for the new issue: 4D-API_cs2cs-style.gie(256)

**Comment by rouault** [comment]
@sebastic First test failure fixed per edfa2981326ea4a476d420436323d0fb37a4efab
Second failure was unrelated to the GCC version but to the lack of the conus grid. Fixed per 666efb98088c007fc297e8cbd0648367bd62c014

**Comment by sebastic** [comment]
Confirmed fixed with the changes from edfa298, thanks!
